docs/root/intro/arch_overview/ip_transparency.rst

```diff
@@ -69,3 +71,25 @@ Some drawbacks to the Original Source filter:
 * It requires that Envoy have access to the downstream remote address.
 * Its configuration is relatively complex.
 * It may introduce a slight performance hit due to restrictions on connection pooling.
+
+.. _arch_overview_ip_transparency_original_src_http:
+
+Original Source HTTP Filter
+---------------------------
+
+In controlled deployments, it may be possible to replicate the downstream remote address on the
+upstream connection by using a
+:ref:`Original Source HTTP filter <config_http_filters_original_src>`. This filter operates much like
+the :ref:`Original Src Listener Filter <arch_overview_ip_transparency_original_src_listener>`. The
+main difference is that it can infer the original source address from HTTP headers, which is important
+for cases where a single downstream connection carries multiple HTTP requests from different original
```

Comment:
Out of curiosity when would you expect this to happen? Would there be something in front of Envoy multiplexing requests from multiple downstreams on the same connection?

Reply:
Yeah, that's what I'm thinking. One example would be in an Istio deployment where envoy is acting first as a front proxy, then as a sidecar. There may be one http2 connection between the front proxy and the sidecar serving many http requests.